python/jmxquery/__init__.py

```diff
@@ -50,125 +60,156 @@ def __init__(self,
         """
-        self.connection_uri = connection_uri
-        self.jmx_username = jmx_username
-        self.jmx_password = jmx_password
-        self.java_path = java_path
 
-    def __run_jar(self, queries: List[JMXQuery], timeout) -> List[JMXQuery]:
+        self.connection_uri = uri
+        self.jmx_username = user
+        self.jmx_password = passwd
+        self.java_path = jpath
+
+    def run_jar(self, queries, timeout):
         """
         Run the JAR and return the results
 
         :param query:   The query
-        :return:        The full command array to run via subprocess
+        :return:        The full command array to run via subprocess32
         """
 
-        command =  [self.java_path, '-jar', JAR_PATH, '-url', self.connection_uri, "-json"]
-        if (self.jmx_username):
-            command.extend(["-u", self.jmx_username, "-p", self.jmx_password])
+        command =  [
+            self.java_path,
+            '-jar',
+            JAR_PATH,
+            '-url',
+            self.connection_uri,
+            "-json"]
+
+        if self.jmx_username:
+            command.extend([
+                "-u",
+                self.jmx_username,
+                "-p",
+                self.jmx_password])
 
         queryString = ""
         for query in queries:
             queryString += query.to_query_string() + ";"
 
         command.extend(["-q", queryString])
-        logger.debug("Running command: " + str(command))
+        log.debug("Running command: " + str(command))
+        print("Running command: %s" % str(command))
 
         jsonOutput = "[]"
         try:
-            output = subprocess.run(command,
-                                    stdout=subprocess.PIPE,
-                                    stderr=subprocess.PIPE,
+            output = subprocess32.run(command,
+                                    stdout=subprocess32.PIPE,
+                                    stderr=subprocess32.PIPE,
                                     timeout=timeout,
                                     check=True)
 
             jsonOutput = output.stdout.decode('utf-8')
-        except subprocess.TimeoutExpired as err:
-            logger.error("Error calling JMX, Timeout of " + str(err.timeout) + " Expired: " + err.output.decode('utf-8'))
-        except subprocess.CalledProcessError as err:
-            logger.error("Error calling JMX: " + err.output.decode('utf-8'))
-            raise err
-
-        logger.debug("JSON Output Received: " + jsonOutput)
-        metrics = self.__load_from_json(jsonOutput)
+        except subprocess32.TimeoutExpired as err:
+            log.error("Timeout of %s Expired: %s" % (
+                str(err.timeout),
+                err))
+
+        except subprocess32.CalledProcessError as err:
+            log.error("Error calling JMX: %s" % (
+                err))
+
+            raise
+
+        except Exception as err:
+            log.error("Exception: %s" % err)
+            raise
```

Comment:
All three of exception handlers do not add any useful information or functionality and should be removed.  In fact, the `TimeoutExpired` error is worse because, by not re-raising the exception, it prevents client code from making a decision about how to handle the situation.

Reply:
I'll remove the try/except clause per your recommendation and leave the responsibility to the client code.

Reply:
Implemented per your recommendation.